awx/main/tasks/jobs.py

```diff
@@ -2148,7 +1984,10 @@ class RunSystemJob(BaseTask):
 
     model = SystemJob
     event_model = SystemJobEvent
-    event_data_key = 'system_job_id'
+    callback_class = RunnerCallbackForSystemJob
+
+    def __init__(self):
+        self.runner_callback = self.callback_class()
```

Comment:
Highest priority still TODO items seems like it would be to delete these `__init__` methods like this.